tests/RobotFramework/tests/tedge_write.robot

```diff
@@ -33,53 +67,93 @@ Preserves permissions if a destination exists
 
 Preserves permissions if a destination exists
     [Arguments]          ${umask}
 
-    ${dest_filename}=    Execute Command    mktemp    strip=True
-    Execute Command      chown tedge:tedge ${dest_filename}
-    Execute Command      chmod 666 ${dest_filename}
+    ${dest_file}=    Create a temporary file
+    Execute Command      chown tedge:tedge ${dest_file}
+    Execute Command      chmod 666 ${dest_file}
+
+    Execute Command as user    tedge
+    ...    umask ${umask} && echo abc | sudo tedge-write ${dest_file} --user root --group root --mode 700
+    
+    Should Have File Ownership    ${dest_file}    tedge:tedge
+    Should Have File Mode         ${dest_file}    666
 
-    Execute Command      umask ${umask} && echo abc | sudo tedge-write ${dest_filename} --user root --group root --mode 700
-    Execute Command      ls ${dest_filename}
-    ${user_owner}=       Execute Command    stat -c '%U' ${dest_filename}    strip=True
-    Should Be Equal      ${user_owner}    tedge
-    ${group_owner}=      Execute Command    stat -c '%G' ${dest_filename}    strip=True
-    Should Be Equal      ${group_owner}    tedge
-    ${mode}=      Execute Command    stat -c '%a' ${dest_filename}    strip=True
-    Should Be Equal      ${mode}    666
+    There should be no leftover temporary files    ${dest_file}
 
 Custom Setup
     Setup    skip_bootstrap=True
     Execute Command    ./bootstrap.sh --no-bootstrap --no-connect
-    Execute Command    sudo --user\=tedge bash
+
+    Create a regular user
+
+Create a temporary file
+    [Arguments]    ${dry_run}=${False}
+    [Documentation]
+    ...    Creates a temporary directory and creates a temporary file in that directory. This is so we can check that
+    ...    after writing there are no leftover temporary files in the destination file's parent directory.
+    ${dir}=     Execute Command    mktemp --directory    strip=True
+    ${path}=    Execute Command    mktemp --tmpdir\=${dir} ${{'--dry-run' if ${dry_run} is True else ''}}   strip=True
+    RETURN    ${path}
+
+Make file inaccessible to regular users
+    [Arguments]    ${file}
+    Execute Command    chown root:root ${file}
+    Execute Command    chmod 600 ${file}
+
+Write content to file using sudo tedge-write as user
+    [Arguments]    ${user}    ${file}    &{named}
+    Execute Command as user    ${user}    echo new content | sudo tedge-write ${file}    &{named}
+
+Create a regular user
+    Execute Command    useradd -s /bin/bash ${REGULAR_USER}
+
+Execute Command as user
+    [Arguments]    ${user}    ${command}    &{named}
+    Execute Command    sudo -n --user\=${user} bash -c '${command}'    &{named}
+
+Should Have File Ownership
+    [Arguments]    ${file}    ${expected_permissions}
+    ${file_mode_ownership}=    Execute Command    stat -c '%U:%G' ${file}    strip=${True}
+    Should Be Equal    ${file_mode_ownership}    ${expected_permissions}    msg=Unexpected file ownership of ${file}
+
+Should Have File Mode
```

Comment:
... and for this one:

```
Should Have File Mode
    [Arguments]    ${file}    ${expected_mode}
    ${actual_mode}=    Execute Command    stat -c '%a' ${file}    strip=${True}
    Should Be Equal    ${actual_mode}    ${expected_mode}    
    ...    msg=Expected permissions of file '${file}' to be '${expected_mode}', but found '${actual_mode}'.

Reply:
Replaced with a keyword from DeviceLibrary in 66cdcc7